tests/network/sriov.go

```diff
@@ -92,514 +88,453 @@ var _ = Describe("[Serial]SRIOV", func() {
-				Expect(err).ToNot(HaveOccurred())
+			By("checking virtual machine instance has an interface with the requested MAC address")
+			ifaceName, err := findIfaceByMAC(virtClient, vmi, mac, 140*time.Second)
+			Expect(err).NotTo(HaveOccurred())
+			vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &k8smetav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(checkMacAddress(vmi, ifaceName, mac)).To(Succeed())
+
+			By("checking virtual machine instance reports the expected network name")
+			Expect(getInterfaceNetworkNameByMAC(vmi, mac)).To(Equal(sriovnet1))
+			By("checking virtual machine instance reports the expected info source")
+			networkInterface := vmispec.LookupInterfaceStatusByMac(vmi.Status.Interfaces, mac)
+			Expect(networkInterface).NotTo(BeNil(), "interface not found")
+			Expect(networkInterface.InfoSource).To(Equal(vmispec.InfoSourceDomainAndGA))
+		})
+
+		Context("migration", func() {
 
-				metadataStruct := cloudinit.ConfigDriveMetadata{
-					InstanceID:   fmt.Sprintf("%s.%s", vmi.Name, vmi.Namespace),
-					InstanceType: testInstancetype,
-					Hostname:     dns.SanitizeHostname(vmi),
-					UUID:         string(vmi.Spec.Domain.Firmware.UUID),
-					Devices:      &deviceData,
+			BeforeEach(func() {
+				if err := validateSRIOVSetup(virtClient, sriovResourceName, 2); err != nil {
+					Skip("Migration tests require at least 2 nodes: " + err.Error())
 				}
+			})
 
-				buf, err := json.Marshal(metadataStruct)
-				Expect(err).ToNot(HaveOccurred())
-				By("mouting cloudinit iso")
-				mountCloudInitConfigDrive := tests.MountCloudInitFunc("config-2")
-				mountCloudInitConfigDrive(vmi)
+			BeforeEach(func() {
+				tests.EnableFeatureGate(virtconfig.SRIOVLiveMigrationGate)
+			})
 
-				By("checking cloudinit meta-data")
-				tests.CheckCloudInitMetaData(vmi, "openstack/latest/meta_data.json", string(buf))
+			AfterEach(func() {
+				tests.DisableFeatureGate(virtconfig.SRIOVLiveMigrationGate)
 			})
 
-			It("[test_id:1754]should create a virtual machine with sriov interface", func() {
-				vmi := getSriovVmi([]string{sriovnet1}, defaultCloudInitNetworkData())
-				vmi = startVmi(vmi)
-				vmi = waitVmi(vmi)
+			var vmi *v1.VirtualMachineInstance
 
-				By("checking KUBEVIRT_RESOURCE_NAME_<networkName> variable is defined in pod")
-				Expect(validatePodKubevirtResourceNameByVMI(virtClient, vmi, sriovnet1, sriovResourceName)).To(Succeed())
+			const mac = "de:ad:00:00:be:ef"
 
-				checkDefaultInterfaceInPod(vmi)
+			BeforeEach(func() {
+				// The SR-IOV VF MAC should be preserved on migration, therefore explicitly specify it.
+				vmi = newSRIOVVmi([]string{sriovnet1}, defaultCloudInitNetworkData())
+				vmi.Spec.Domain.Devices.Interfaces[1].MacAddress = mac
 
-				By("checking virtual machine instance has two interfaces")
-				checkInterfacesInGuest(vmi, []string{"eth0", "eth1"})
+				vmi, err := createVMIAndWait(vmi)
```

Comment:
This has now shadowed the vmi variable, causing the test not to get this last update.

Reply:
Fixed